Docker config add xdebug capability

dnsl48 opened the pull request:
Adds xdebug capability to the docker container so it's debuggable by a localhost IDE.

The debug mode should work seamlessly for 127.0.0.1:9000 by running `./cow/docker/dbg` script (syntactically identical to `./cow/docker/run` and to `./cow/bin/cow`)

bergice commented:
Doesn't work on Mac.

![image](https://user-images.githubusercontent.com/541886/70491014-d4daa580-1b64-11ea-896c-8900db9cac5a.png)

dnsl48 commented:
Fair enough. I think we've found an issue with XDebug not connecting on MacOS (its localhost isn't attached to the host machine since docker on macs runs within a VM). We may need to change the xdebug settings there to connect to `host.docker.internal` rather than `127.0.0.1`. Just need some time to refactor the configuration bit (as @bergice suggested it would be cool to make it configurable).

> comments about providing docs to help people understand the purpose of each command

Sounds reasonable. I think it's worth writing some docs when the concept is approved (I guess it is already, so I'll write some docs :)

> At first glance it looks sensible and I'm happy to take you at your word that this will be helpful.

well, I'm already using it for debugging myself, it works and it's helpful :)

dnsl48 commented:
- Added shim fixes for MacOS X compatibility
 - Added documentation
 - Added XDebug configs control through `.env` file
 
Ready for another review

bergice commented on docker/README.md:
```diff
@@ -0,0 +1,46 @@
+(such as git, composer, ruby gems and transifex client).
+
+`docker-compose.yml` contains a set of configurations for different use cases such as running cow for development, debugging, running tests, debugging tests, running code sniffer and running cow in production mode.
+Although you may run any of these configs straightforwardly by runnig docker-compose if you want, the idea is to run them
```
```suggestion
Although you may run any of these configs straightforwardly by running docker-compose if you want, the idea is to run them
```

bergice commented on docker/README.md:
```diff
@@ -0,0 +1,46 @@
+Running `./bin/cow` transparently within a container.
+
+This script runs cow in development mode by mounting current folder (`$(pwd)`) into the container preserving the path.
+It also uses the current cow folder as is, so if you patched any scripts, the changes will be runnnig exactly as is within the container.
```
```suggestion
It also uses the current cow folder as is, so if you patched any scripts, the changes will be running exactly as is within the container.
```
